app/js/components/queryBuilder/queryBuilder.component.js

```diff
@@ -124,6 +126,40 @@ const QueryBuilderComponent = {
 
 
 
+                $rootScope.$on('query:submit', ()=>{
+                  let diseases = [];
+                  let genes = [];
+
+                  // Check to make sure we have at least 1 gene and 1 disease
+                  if(this.diseaseList.length > 0 && this.mutationList.length > 0) {
+                    diseases = this.diseaseList.map(function(obj) {
+                      return obj['acronym'];
+                    });
+
+                    genes = this.mutationList.map(function(obj) {
+                      return obj['entrezgene'];
+                    });
+
+                    let query = $resource(`${AppSettings.api.baseUrl}${AppSettings.api.classifiers}/`, {}, {
+                      submit: {
+                        method: 'POST',
+                        headers: {
+                          Authorization: 'Bearer tked28ic74ebiyy6bd9b6mqxa',
```

Comment:
The random_slug here shouldn't be hardcoded into the request. I believe it should be the random_slug of the user accessing the site.